components/webext-storage/src/store.rs

```diff
@@ -116,6 +117,16 @@ impl Store {
     pub fn close(self) -> result::Result<(), (Store, Error)> {
         self.db.close().map_err(|(db, err)| (Store { db }, err))
     }
+
+    /// Migrates data from a database in the format of the "old" kinto
+    /// implementation. Returns the count of webextensions for whom data was
+    /// migrated.
+    pub fn migrate(self, filename: &str) -> Result<usize> {
```

Comment:
I'd like this to return more information. In particular:

- number of total records we tried to migrate
- number of records successfully migrated
- if we had any recoverable internal errors
- whether not any of the records overflowed quotas.